Userland/Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -464,16 +461,15 @@ static inline ErrorOr<Marker> read_marker_at_cursor(InputMemoryStream& stream)
         return JPG_INVALID;
     u8 next;
     do {
-        stream >> next;
-        TRY(stream.try_handle_any_error());
+        next = TRY(stream.read_value<u8>());
         if (next == 0x00)
             return JPG_INVALID;
     } while (next == 0xFF);
     marker = 0xFF00 | (u16)next;
     return is_valid_marker(marker) ? marker : JPG_INVALID;
 }
 
-static ErrorOr<void> read_start_of_scan(InputMemoryStream& stream, JPGLoadingContext& context)
+static ErrorOr<void> read_start_of_scan(Core::Stream::FixedMemoryStream& stream, JPGLoadingContext& context)
```

Comment:
Is there a particular reason that you have chosen `Core::Stream::FixedMemoryStream` here? Otherwise, we'd probably rather want one of the abstract classes, most likely `Core::Stream::Stream`.

Reply:
I used it because it has an `offset()` function and the loader uses it to print the byte offsets when a decoding failed under JPG_DEBUG macro flag. Should we get rid of it?

Reply:
If you want to keep printing the offset, we could replace it with `SeekableStream` and use `tell()` instead.